### PR TITLE
Introduce `ToBackend` expression

### PR DESCRIPTION
**Pull request by rjzamora**
Introduces a simple `ToBackend` (and `ToPandasBackend` implementation).

The current approach of using `map_partitions` in `PandasBackendEntrypoint` effectively blocks query-planning optimizations when data is moved between GPU and CPU. The obvious solution is to use a simple `Expr` class.

### CODE REVIEW

**Comment by rjzamora** [comment]
@phofl - This PR is (hopefully) much less controversial than the resource barrier idea in #1116 :)

File: dask_expr/_backends.py [hunk 1]
```diff
@@ -32,25 +33,31 @@
 )
 
 
+class ToPandasBackend(ToBackend):
+    @staticmethod
+    def operation(df, options):
+        from dask.dataframe.dispatch import to_pandas_dispatch
```
Comment (phofl) [inline dask_expr/_backends.py:39]: any reason this is a local import? we import tons of stuff from dask.dataframe globally

File: dask_expr/_expr.py [hunk 2]
```diff
@@ -1303,6 +1303,17 @@ def operation(df):
...
 
+class ToBackend(Elemwise):
+    _parameters = ["frame", "options"]
+    _projection_passthrough = True
+    _filter_passthrough = True
+    _preserves_partitioning_information = True
+
+    @staticmethod
+    def operation(df, options):
+        raise NotImplementedError()
```
Comment (phofl) [inline dask_expr/_expr.py:1314]: Can you set operation = None, this should raise during graph construction, not execution IIRC

File: dask_expr/_backends.py (hunk 1 above)
Comment (rjzamora) [reply dask_expr/_backends.py:39]: Nope - I'll move it to the top level

File: dask_expr/_expr.py (hunk 2 above)
Comment (rjzamora) [reply dask_expr/_expr.py:1314]: Sounds good

Comment (rjzamora) [reply dask_expr/_expr.py]: Okay - Looks like `Elemwise` already has `operation = None` by default. So, I just removed the operation def altogether. Thanks!

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**